main/src/main/scala/sbt/internal/server/BuildServerProtocol.scala

```diff
@@ -186,6 +186,23 @@ object BuildServerProtocol {
     bspBuildTargetCompile / aggregate := false,
     bspBuildTargetTest := bspTestTask.evaluated,
     bspBuildTargetTest / aggregate := false,
+    bspBuildTargetCleanCache := Def.inputTaskDyn {
+      val s: State = state.value
+      val targets = spaceDelimited().parsed.map(uri => BuildTargetIdentifier(URI.create(uri)))
+      val workspace = bspFullWorkspace.value.filter(targets)
+      workspace.warnIfBuildsNonEmpty(Method.Compile, s.log)
+      val filter = ScopeFilter.in(workspace.scopes.values.toList)
+      Def.task {
+        val statusCodes = Keys.clean.result.all(filter).value
```

Comment:
I think `result` is a better name than `statusCodes` because `statusCodes` generally means there is an `Int` somewhere describing the output of the task. But here the type is `Seq[Result[Unit]]`.

```suggestion
        val results = Keys.clean.result.all(filter).value
```